File: Data Structures & Algorithms/permutation-string/submission-0.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
        vector<int> v1(26,0), v2(26,0);
        for(int i=0;i<s1.size();i++){
            v1[s1[i]-'a']++;
        }

        for(int i=0;i<s2.size();i++){
            v2[s2[i]-'a']++;

            if(i>=s1.size()){
                v2[s2[i-s1.size()]-'a']--;
            }

            if(v1==v2){
                return true;
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/merge-k-sorted-linked-lists/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode mergeKLists(ListNode[] lists) {
        ListNode dummy=new ListNode(-1);
        ListNode curr=dummy;

        PriorityQueue<ListNode> pq= new PriorityQueue<>(
            (a,b)-> a.val - b.val
        );
        int l=lists.length;
        int j=0;
        
        if(l==0){
            return null;
        }
        for(int i=0;i<l;i++){
            if(lists[i]!=null)
                pq.offer(lists[i]);
        }

        while(!pq.isEmpty()){
            ListNode small=pq.poll();
            curr.next= small;;
            curr=curr.next;

            if(small.next!=null){
                pq.offer(small.next);
            }

        }
        return dummy.next;
    }
}
